cmd/soroban-cli/src/config/key.rs

```diff
@@ -0,0 +1,148 @@
+    MuxedAccount { muxed_account: MuxedAccount },
+}
+
+impl Key {
+    pub fn public_key(&self, hd_path: Option<usize>) -> Result<xdr::MuxedAccount, Error> {
+        let bytes = match self {
+            Key::Secret(secret) => secret.public_key(hd_path)?.0,
+            Key::PublicKey {
+                public_key: Public(key),
+            } => key.0,
+            Key::MuxedAccount {
+                muxed_account: MuxedAccount(stellar_strkey::ed25519::MuxedAccount { ed25519, id }),
+            } => {
+                return Ok(xdr::MuxedAccount::MuxedEd25519(xdr::MuxedAccountMed25519 {
+                    ed25519: xdr::Uint256(*ed25519),
+                    id: *id,
+                }))
+            }
+        };
+        Ok(xdr::MuxedAccount::Ed25519(xdr::Uint256(bytes)))
+    }
+
+    pub fn private_key(
+        &self,
+        hd_path: Option<usize>,
+    ) -> Result<stellar_strkey::ed25519::PrivateKey, Error> {
+        match self {
+            Key::Secret(secret) => Ok(secret.private_key(hd_path)?),
+            _ => Err(Error::SecretPublicKey),
+        }
+    }
+}
+
+impl FromStr for Key {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if let Ok(secret) = s.parse() {
+            return Ok(Key::Secret(secret));
+        }
+        if let Ok(public_key) = s.parse() {
+            return Ok(Key::PublicKey { public_key });
+        }
+        if let Ok(muxed_account) = s.parse() {
+            return Ok(Key::MuxedAccount { muxed_account });
+        }
+        Err(Error::Parse(s.to_owned()))
+    }
+}
+
+impl From<stellar_strkey::ed25519::PublicKey> for Key {
+    fn from(value: stellar_strkey::ed25519::PublicKey) -> Self {
+        Key::PublicKey {
+            public_key: Public(value),
+        }
+    }
+}
+
+impl From<&stellar_strkey::ed25519::PublicKey> for Key {
+    fn from(stellar_strkey::ed25519::PublicKey(key): &stellar_strkey::ed25519::PublicKey) -> Self {
+        stellar_strkey::ed25519::PublicKey(*key).into()
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, serde_with::SerializeDisplay, serde_with::DeserializeFromStr)]
+pub struct Public(pub stellar_strkey::ed25519::PublicKey);
+
+impl FromStr for Public {
+    type Err = stellar_strkey::DecodeError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(Public(stellar_strkey::ed25519::PublicKey::from_str(s)?))
+    }
+}
+
+impl Display for Public {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl From<&Public> for stellar_strkey::ed25519::MuxedAccount {
+    fn from(Public(stellar_strkey::ed25519::PublicKey(key)): &Public) -> Self {
+        stellar_strkey::ed25519::MuxedAccount {
+            id: 0,
+            ed25519: *key,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, serde_with::SerializeDisplay, serde_with::DeserializeFromStr)]
+pub struct MuxedAccount(pub stellar_strkey::ed25519::MuxedAccount);
```

Comment:
I don't think this type is required. FromStr and Display are implemented for MuxedAccount:

https://github.com/stellar/rs-stellar-xdr/blob/715003372ea6380044b5a4a02907ff73e56ae9e7/src/curr/str.rs#L102-L120

Same with Public.

Reply:
It's the `serde_with` part which requires the type, allowing serialization via those traits. We could remove this once this is released: https://github.com/stellar/rs-stellar-strkey/pull/72